.github/workflows/build_and_test_pr.yml

```diff
@@ -0,0 +1,195 @@
+          python-arch: x64
+          generator: "Visual Studio 17 2022"
+          nproc: 4
+          targets: energyplus ExpandObjects_build ReadVars_build Slab_build Basement_build AppGPostProcess_build ParametricPreprocessor_build
+          run_regressions: true
+
+    steps:
+
+    - name: Set up Python ${{ env.Python_REQUIRED_VERSION }}
+      id: setup-python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ env.Python_REQUIRED_VERSION }}
+
+    - name: Install Dependencies for Mac
+      if: ${{ runner.os == 'macOS' }}
+      run: |
+        brew update
+        brew install gcc@13
+        echo "FC=$(brew --prefix gcc@13)/bin/gfortran-13" >> $GITHUB_ENV
+        echo MACOSX_DEPLOYMENT_TARGET=${{ matrix.macos_dev_target }} >> $GITHUB_ENV
+
+    - name: Install Dependencies for Linux
+      if: ${{ runner.os == 'Linux' }}
+      run: |
+        sudo apt-get update
+        sudo apt-get install libxkbcommon-x11-0 xorg-dev libgl1-mesa-dev
+        if [[ "${{ matrix.os }}" == "ubuntu-24.04" ]]; then
+          # https://github.com/actions/runner-images/issues/10025
+          echo "FC=gfortran-13" >> $GITHUB_ENV
+        fi
+
+    # BUILD AND TEST INTEGRATION FILES ON THE BASELINE BRANCH
+
+    - name: Baseline Checkout
+      if: ${{ matrix.run_regressions }}
+      uses: actions/checkout@v4
+      with:
+        ref: develop
+        path: baseline
+
+    - name: Baseline Create Build Directory
+      if: ${{ matrix.run_regressions }}
+      run: cmake -E make_directory ./baseline/build/
+
+    - name: Baseline Configure CMake
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      run: >
+        cmake 
+        -G "${{ matrix.generator }}"
+        -DCMAKE_BUILD_TYPE:STRING=Release
+        -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${{ matrix.macos_dev_target }}
+        -DLINK_WITH_PYTHON:BOOL=ON
+        -DPython_REQUIRED_VERSION:STRING=${{ steps.setup-python.outputs.python-version }}
+        -DPython_ROOT_DIR:PATH=$RUNNER_TOOL_CACHE/Python/${{ steps.setup-python.outputs.python-version }}/${{ matrix.python-arch }}/
+        -DBUILD_TESTING:BOOL=ON
+        -DBUILD_FORTRAN:BOOL=ON 
+        -DBUILD_PACKAGE:BOOL=OFF
+        -DDOCUMENTATION_BUILD:STRING=DoNotBuild
+        -DENABLE_OPENMP:BOOL=OFF
+        -DUSE_OpenMP:BOOL=OFF
+        ../
+
+    - name: Baseline Build
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      # Building specific target list so that we don't waste time building the unit test binary
+      run: cmake --build . -j ${{ matrix.nproc }} --config Release --target ${{ matrix.targets }}
+
+    - name: Baseline Test
+      if: ${{ matrix.run_regressions }}
+      working-directory: ./baseline/build
+      run: ctest -C Release -E Basement -R integration -j 3  # TODO: Speed up basement so we don't have to skip it.
+
+    # BUILD AND TEST EVERYTHING ON THE CURRENT BRANCH
+
+    - name: Branch Checkout
```

Comment:
Regardless of whether we are doing regressions, we will always build the branch and run the full test suite on it.